File: hackerrank/warmup/utopian-tree/utopian-tree.py
#!/usr/bin/env python

def size(cycles):
  size = 1
  for i in xrange(cycles):
    if (i % 2) != 0:
      size = size + 1
    else:
      size = size * 2
  return size

inputNumber = int(raw_input())

for i in xrange(inputNumber):
  cycles = int(raw_input())
  print size(cycles)

File: hackerrank/warmup/utopian-tree/utopian-tree.rb
#!/usr/bin/env ruby

inputNumber = gets.to_i

(1..inputNumber).each {
  cycles = gets.to_i

  size = 1;
  (1..cycles).each {|i|
    size = (i % 2 != 0) ? size * 2 : size + 1
  }

  puts size
}
